web3/src/main/java/org/hiero/mirror/web3/controller/GenericControllerAdvice.java

```diff
@@ -78,9 +99,16 @@ private ResponseEntity<?> mirrorEvmTransactionError(
         final var childTransactionErrors = e.getChildTransactionErrors().stream()
                 .map(message -> new ErrorMessage(message, StringUtils.EMPTY, StringUtils.EMPTY))
                 .toList();
-        return new ResponseEntity<>(
-                new GenericErrorResponse(e.getMessage(), e.getDetail(), e.getData(), childTransactionErrors),
-                BAD_REQUEST);
+
+        if (e.getResponseCode() != null && SERVER_RESPONSE_CODES.contains(e.getResponseCode())) {
```

Comment:
The set returns false for null.

```suggestion
        if (SERVER_RESPONSE_CODES.contains(e.getResponseCode())) {
```

Reply:
Fixed